src/test/java/com/hivemq/extensions/handler/ConnackOutboundInterceptorHandlerTest.java

```diff
@@ -119,7 +122,12 @@ public void setUp() throws Exception {
         pluginTaskExecutorService = new PluginTaskExecutorServiceImpl(() -> executor1, mock(ShutdownHooks.class));
 
         handler = new ConnackOutboundInterceptorHandler(configurationService, asyncer, hiveMQExtensions, pluginTaskExecutorService, interceptors, serverInformation, eventLog);
-        channel.pipeline().addFirst(handler);
+        channel.pipeline().addLast("test", new ChannelOutboundHandlerAdapter() {
```

Comment:
just add an InterceptorHandler here, you can do this in every other test, too. So you can also test your new Handler. Just create an TestOutbound(Inbound)InterceptorHandler, which has the all necessary dependencies as instances.